tensorrt_llm/_torch/pyexecutor/kv_cache_manager_v2.py

```
@@ -642,6 +643,39 @@ def _augment_tokens_with_contiguous_mm_metadata(
+    """
+    # Integer division also covers a partial (mid-block) match: that block is
+    # still the first one this sequence completes and commits.
+    block_index = num_reusable_tokens // tokens_per_block
+    num_tokens_needed = (block_index + 1) * tokens_per_block
+    if num_tokens_needed > len(tokens):
+        return None
+    # A block key depends only on the tokens preceding it, so hashing the
+    # truncated prefix is exact -- and keeps both the token marshalling and the
+    # hash chain proportional to the block we want, not to the whole prompt.
+    key = None
+    for _, key in sequence_to_blockchain_keys(
```

Ruff (0.16.2)

[warning] Loop control variable `key` not used within loop body

(B007)